src/io/reader.rs

```diff
@@ -225,7 +226,7 @@ impl<R: BufRead + Seek> Reader<R> {
     /// If no format was determined, returns an `ImageError::Unsupported`.
     pub fn decode(mut self) -> ImageResult<DynamicImage> {
         let format = self.require_format()?;
-        free_functions::load_inner(self.inner, self.limits, format)
+        free_functions::load_inner(self.inner, self.limits, format, LoadErrorHandling::Strict)
     }
```

Comment:
Preferrably you'd add `fn decode_lossy` for `Reader` instead of `image::load_lossy`. Use of the module functions is somewhat discouraged due to lack of limit handling and imprecise error reporting.